src/CQRS/LeanCode.CQRS.Security/UserIdExtractor.cs

```diff
@@ -0,0 +1,30 @@
+using System.Security.Claims;
+
+namespace LeanCode.CQRS.Security;
+
+public interface IUserIdExtractor<TUserId> : IUserIdExtractor
+    where TUserId : notnull, IEquatable<TUserId>
+{
+    new TUserId Extract(ClaimsPrincipal user);
```

Comment:
Let's not use shadowing. These are both valid methods and both should be exposed.

Reply:
https://leancode.slack.com/archives/CH27S36HZ/p1701867170839379 - discussion about it